### ISSUE TITLE
Document.getFilePropertiesAsync method fails in new Excel document

### ISSUE THREAD

**Issue by Kevin-Wallace-Uptime** [OfficeDev/office-js#3509]
## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC and Mac
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: PC- Version 2302, Mac- 16.75
* Operating System: PC- Windows 11, Mac- Monterey
* Browser (if using Office on the web):N/A
 
## Expected behavior
In the event that a new workbook has not been saved, the call to Document.getFilePropertiesAsync should be successful and return value as an empty string. This is how it works in both PowerPoint and Word. For reference Word and PowerPoint return {value: "", status: "succeeded"}

## Current behavior
When calling the method Document.getFilePropertiesAsync, it returns {value: undefined, status: "failed", error: }

## Steps to reproduce

1. Open a new blank Excel Document (the document must NOT be saved)
2. run this gist via Script Lab https://gist.github.com/Kevin-Wallace-Uptime/b9badf7f00b146720e29d5c63a4fb356
3. Click the run button
4. the console will report the method failed 

## Link to live example(s)
1. [Script Lab Gist](https://gist.github.com/Kevin-Wallace-Uptime/6dd0df3c31e24c19e0d93df95c03598d)
Just in case you can't import the gist, here it is
```
$("#run").click(() => tryCatch(run));

async function run() {
  Office.context.document.getFilePropertiesAsync({}, (res) => { 
    console.log(res) 
  })
}

/** Default helper for invoking an action and handling errors. */
async function tryCatch(callback) {
  try {
    await callback();
  } catch (error) {
    // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
    console.error(error);
  }
}
```

# Provide additional details

## Context
This behavior makes handling a failure less clear. Did it happen because it is a new unsaved Workbook or did it fail for some other reason?

## Useful logs
- [x] Console errors
- [x] Screenshots
- [ ] Test file (if only happens on a particular file)
<img width="1388" alt="image" src="https://github.com/OfficeDev/office-js/assets/131808660/f6a1a310-2dab-4160-a76a-7bf006892bde">
<img width="1719" alt="image" src="https://github.com/OfficeDev/office-js/assets/131808660/5d855fad-9daf-426b-8fd2-002cefe4d812">

**Comment by ZYUN-MSFT** [comment]
Hi @Kevin-Wallace-Uptime ,

Thanks for reaching us. We'd like to know which scenario you are using this API?\
Since the FileProperties are usually supposed to be read within a saved exist Excel file.

Thanks.

**Comment by Kevin-Wallace-Uptime** [comment]
> Since the FileProperties are usually supposed to be read within a saved exist Excel file.

I guess it is usually a saved existing Excel file, but shouldn't it be supported in a new unsaved excel file?
Because calling `getFilePropertiesAsync` in an unsaved Excel generates an error.  When calling the same function in an unsaved Word document, or an unsaved PowerPoint Presentation there is NOT an error.  This behavior is inconsistent across Hosts/Applications and seems to be an Excel-specific bug in office-js.

**Comment by ZYUN-MSFT** [comment]
Hi @Kevin-Wallace-Uptime ,

We have created a internal BUG 8121169 in our backlog to track this.\
But we'd want to know that whether the related fucntion is critical in your scenario?\
Is it a blocking issue or there have a workaround of this?

Thanks.

**Comment by Kevin-Wallace-Uptime** [comment]
@ZYUN-MSFT thank you! I do have a workaround for the time being.

**Comment by UMAMAHESHWARIG** [comment]
Hi team, 
we also have the same issue. 
what is the update on the bug

**Comment by UMAMAHESHWARIG** [comment]
[Kevin-Wallace-Uptime](https://github.com/Kevin-Wallace-Uptime) what is the work around you are using ?
it might help us as we are also getting the same error

**Comment by Kevin-Wallace-Uptime** [comment]
@UMAMAHESHWARIG I basically do getFilePropertiesAsync, and if the callback returns status as failed, I check if the add in is running in Excel, and if `Office.context.document.url` is an empty string. It's obviously been awhile but I believe I tested this approach pretty extensively and never found an issue with the workaround. Here is a snippet:
```
Office.context.document.getFilePropertiesAsync((res) => {
  if (res.status === Office.AsyncResultStatus.Succeeded) {
    // success. no need to handle special case for unsaved document
  } else {
    // for some reason a new unsaved excel document never calls this successfully
    // Link to issue I opened https://github.com/OfficeDev/office-js/issues/3509
    if (
      Office.context.diagnostics.host === Office.HostType.Excel &&
      Office.context.document.url === ""
    ) {
      // can handle unsaved excel document here
      // I think I just give the document a default name like "Book1.xlsx"
    }
    // throw some error
    reject("could not get file name");
  }
});
```

**Comment by jaiswalshubh1** [comment]
Hi @Kevin-Wallace-Uptime ,

This issue has been fixed, Can you please verify from your side?  

cc: @kustonaut 

Thanks!
Shubham Jaiswal

**Comment by Kevin-Wallace-Uptime** [comment]
@jaiswalshubh1 Yes this appears to be fixed. I ran the gist that I posted in the original and it now returns: 
```
{
    "value": {
        "url": ""
    },
    "status": "succeeded"
}
```

I believe this is the expected behavior. Thanks for the fix.

**Comment by jaiswalshubh1** [comment]
Thank you @Kevin-Wallace-Uptime for the confirmation, I am closing this issue.

CC: @kustonaut
